### PR TITLE
Fix `closeAllComponents did not finish on time` problem at the pruning storer level

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request
- During heavy testing we observed that the pruning stores were not closing correctly while the node carried a snapshot.
  
## Proposed Changes
- added a forgotten mutex on the `Close()` method in the pruning storer.
- added more relevant prints for easier debugging

## Testing procedure
- internal system testnets

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -1034,7 +1034,14 @@ func (n *Node) Close() error {
...
 	var closeError error = nil
-	log.Debug("closing all managed components")
+
+	allComponents := make([]string, 0, len(n.closableComponents))
+	for i := len(n.closableComponents) - 1; i >= 0; i-- {
+		managedComponent := n.closableComponents[i]
+		allComponents = append(allComponents, fmt.Sprintf("%v", managedComponent))
+	}
+
+	log.Debug("closing all managed components", "all components", strings.Join(allComponents, ", "))
```
Comment (iulianpascalau) [inline node/node.go:1044]: maybe rephrase this? Add the "order" perhaps?

File: node/nodeRunner.go [hunk 2]
```diff
@@ -823,7 +823,12 @@ func waitForSignal(
 	case <-chanCloseComponents:
 		log.Debug("Closed all components gracefully")
 	case <-time.After(maxTimeToClose):
-		log.Warn("force closing the node", "error", "closeAllComponents did not finish on time")
+		buff := make([]byte, 100*1024*1024)
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/nodeRunner.go:826]: done

File: node/node.go (hunk 1 above)
Comment (iulianpascalau) [reply node/node.go:1044]: done

**Review by raduchis — APPROVED** [review]

File: common/goRoutines/dump.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package goRoutines
```
Comment (bogdan-rosianu) [inline common/goRoutines/dump.go:1]: please: `goroutines`

**Review by sstanculeanu — APPROVED** [review]

File: storage/pruning/pruningStorer_test.go [hunk 4]
```diff
@@ -986,7 +986,7 @@ func TestPruningStorer_processPersistersToClose(t *testing.T) {
 }
 
 func TestPruningStorer_ConcurrentOperations(t *testing.T) {
-	t.Skip("this test should be run only when troubleshooting pruning storer concurrent operations")
+	numOperations := 100 // increase this to 5000 when troubleshooting pruning storer concurrent operations
```
Comment (bogdan-rosianu) [inline storage/pruning/pruningStorer_test.go:989]: since we'll run this test by default, please check that there is no panic on a recover

File: common/goRoutines/dump.go (hunk 3 above)
Comment (iulianpascalau) [reply common/goRoutines/dump.go]: done

File: storage/pruning/pruningStorer_test.go (hunk 4 above)
Comment (iulianpascalau) [reply storage/pruning/pruningStorer_test.go:989]: done

**Review by sstanculeanu — DISMISSED** [review]

File: storage/pruning/pruningStorer_test.go [hunk 5]
```diff
@@ -1012,6 +1012,10 @@ func TestPruningStorer_ConcurrentOperations(t *testing.T) {
 	require.NotNil(t, ps)
 	defer func() {
 		_ = ps.Close()
+		r := recover()
+		if r != nil {
+			assert.Fail(t, fmt.Sprintf("should have not packed %v", r))
```
Comment (bogdan-rosianu) [inline storage/pruning/pruningStorer_test.go:1017]: panicked

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed .
